code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,253 @@
+	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	playsound(A, "alien_claw_flesh", 25, TRUE)
+	X.gain_plasma(X.xeno_caste.drain_plasma_gain)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Rejuvenate/Transfusion
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "rejuvenation"
+	mechanics_text = "When used on self, drains blood and restores health over time. When used on another xenomorph, costs blood and restores some of their health. When used on a dead human, you heal gradually."
+	cooldown_timer = 2 SECONDS
+	plasma_cost = 0
+	use_state_flags = XACT_TARGET_SELF
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/rejuvenate/can_use_ability(atom/target, silent = FALSE, override_flags) //it is set up to only return true on specific xeno or human targets
+	. = ..()
+	if(!.)
+		return
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
+			if(!silent)
+				to_chat(owner, span_notice("We can only use still, dead hosts to heal."))
+			return FALSE
+		else if(!owner.Adjacent(target))
+			if(!silent)
+				to_chat(owner, span_notice("We need to be next to our meal."))
+			return FALSE
+		return TRUE
+
+	if(isxeno(target))
+		var/mob/living/carbon/xenomorph/gorger/X = owner
+		if(X != target)
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				if(!silent)
+					to_chat(X, span_notice("It is beyond our reach, we must be close and our way must be clear."))
+				return FALSE
+			if(X.plasma_stored < GORGER_REJUVENATE_ALLY_COST)
+				if(!silent)
+					to_chat(X, span_notice("We need [GORGER_REJUVENATE_ALLY_COST - X.plasma_stored]u more blood to restore a sister."))
+				return FALSE
+			if(isdead(target))
+				if(!silent)
+					to_chat(X, span_notice("We can only help living sisters."))
+				return FALSE
+			if(!do_mob(X, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+				return FALSE
+			return TRUE
+
+		if(!COOLDOWN_CHECK(X, rejuvenate_self_cooldown))
+			if(!silent)
+				to_chat(X, span_notice("We need another [round(COOLDOWN_TIMELEFT(X, rejuvenate_self_cooldown) / 10)] seconds before we can revitalize ourselves."))
+			return FALSE
+		if(X.plasma_stored < GORGER_REJUVENATE_SELF_COST)
+			if(!silent)
+				to_chat(X, span_notice("We need [GORGER_REJUVENATE_SELF_COST - X.plasma_stored]u more blood to revitalize ourselves."))
+			return FALSE
+		return TRUE
+
+	if(!silent)
+		to_chat(owner, span_notice("We can only drain or restore familiar biological lifeforms."))
+	return FALSE
+
+/datum/action/xeno_action/activable/rejuvenate/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(ishuman(A))
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE))
```

Comment:
No max blood check

Reply:
added check for blood and for defibbable